internal/langserver/handlers/service.go

```diff
@@ -463,6 +463,10 @@ func (svc *service) configureSessionDependencies(ctx context.Context, cfgOpts *s
 			moduleHooks = append(moduleHooks, callRefreshClientCommand(svc.server, commandId))
 		}
 
+		if commandId, ok := lsp.ExperimentalClientCapabilities(cc.Experimental).RefreshTerraformVersionCommandId(); ok {
+			moduleHooks = append(moduleHooks, callRefreshClientCommand(svc.server, commandId))
+		}
```

Comment:
I think in the long run we'll probably want to create a separate hook which only fires when the data exposed through each command changes (not just any data), but this is 🆗  for now, since the other two refresh commands above have the exact same problem.